src/arithmetic/string_funcs/string_funcs.c

```diff
@@ -241,6 +241,83 @@ SIValue AR_ENDSWITH(SIValue *argv, int argc) {
 	return SI_BoolVal(true);
 }
 
+/* returns a string in which all occurrences of a specified string in the original string have been replaced by ANOTHER (specified) string. */
+SIValue AR_REPLACE(SIValue *argv, int argc) {
+	// No string contains null.
+	if(SIValue_IsNull(argv[0]) || SIValue_IsNull(argv[1]) || SIValue_IsNull(argv[1])) return SI_NullVal();
+
+	char *str = strdup(argv[0].stringval);
+	const char *old_string = argv[1].stringval;
+	const char *new_string = argv[2].stringval;
+	size_t old_string_len = strlen(old_string);
+	size_t new_string_len = strlen(new_string);
+
+	if(old_string_len == 0) {
+		ErrorCtx_RaiseRuntimeException("ArgumentError: search string can't be length 0");
+		// Incase expection handler wasn't set, return NULL.
+		return SI_NullVal();
+	}
+
+	char *ptr = str;
+
+	char **arr = array_new(char *, 0);
+
+	while(true) {
+        // find pointer to next substring
+        ptr = strstr(ptr, old_string);
+
+        // if no substring found, then break from the loop
+        if(ptr == NULL) { break; }
+
+		// store ptr for replace use
+		array_append(arr, ptr);
+
+        // increment our string pointer
+        ptr += old_string_len;
+    }
+
+	int occurances = array_len(arr);
+	if(occurances == 0) return argv[0];
```

Comment:
mustn't do that, it is assumed we're returning a new string which will be freed at a later point in time